### DISCUSSION TITLE
Genie special ability rate

### DISCUSSION THREAD

**Discussion by Branikolog** [ihhub/fheroes2#5566] [General]
Upon comparing the rates of genies' special ability from the OG and fheroes2 engine I came into conclusion, that OG provides more chance to half enemy troops.
In fheroes2 engine, I made like 150 hits with genie and the rate was ~10%. (9% actually, but I'm 100% sure it's just a small dispersion). 
While in the OG after more than 200 hits I got average chance about 15% of triggering that special ability.
I also heard, that some different versions of original game have slight differences in AI logic, for instance. Could it be possible, that in some versions of the OG genies' ability has different rate due to mistake? Or is this just mistake of the game manual?
So the main question is should we stick to the original rate (which is higher obviously) more than to 10% of the current implentation?

**Comment by LeHerosInconnu** [comment]
Hello @Branikolog,

> Upon comparing the rates of genies' special ability from the OG and fheroes2 engine I came into conclusion, that OG provides more chance to half enemy troops. In fheroes2 engine, I made like 150 hits with genie and the rate was ~10%. (9% actually, but I'm 100% sure it's just a small dispersion). While in the OG after more than 200 hits I got average chance about 15% of triggering that special ability. I also heard, that some different versions of original game have slight differences in AI logic, for instance. Could it be possible, that in some versions of the OG genies' ability has different rate due to mistake? Or is this just mistake of the game manual? So the main question is should we stick to the original rate (which is higher obviously) more than to 10% of the current implentation?

We've lived with this idea of a 10% chance of triggering his special ability, and even if the manual is wrong, I'm okay with leaving it at 10% especially if it's the AI leading the genies. ;)
On the other hand, 15% could just as easily be 10% or 20% and if in the original game the value is higher than 10%, the player's game experience should remain the same as in the original game.

Of course you could push the test a bit further, say with 1000 attacks :rofl: :1st_place_medal: :eyeglasses: and if it turns out that the genie has a 20% chance of triggering his special ability then it should be corrected.

Wouldn't it be possible to find the exact value in the game source code?

**Reply by ihhub** [reply]
Based on theory of big numbers and probability theory there will be no difference in percentages between 200 and 1000 hits. Even for 10000 it'll remain the the same.

**Reply by Branikolog** [reply]
Since not so many people read manuals and there were no visual description of the ability in game, I think, we could prefer higher rate as players got used to it. On the other hand we should know the exact value. Investigating the original code cannot guarantee us the exact value, since there could be some kind of bug, rising the probability of the genies' ability...
So, we cannot be sure, that 15% rate is a precise value... While testing I've experienced such thing, that staring a new scenario and initiating a battle would lead to the same outcome, especially for the first few hits in a battle...
Also, I would like to know, whether OG versions for different regions experience that higher rate...

**Reply by zenseii** [reply]
I believe 10 % was chosen for fheroes2 because it was reported in the manual.

I see good arguments for both so for me it doesn't really. 

One point that hasn't been raised for keeping the current logic is the point about balancing, but since we don't want to try to rebalance the game. It might just be logical to use the in game probability since that is most likely the balance the game was playtested against.

**Reply by ihhub** [reply]
Based on my data and observations it is exactly 20%.
